File: leetcode-solutions/0268-missing-number/solution.java
class Solution {
    public int missingNumber(int[] arr) {
        int n = arr.length;
        int xor=0;
        int xorArr=0;
        for(int num:arr){
            xorArr^=num;
        }
        for(int i=1;i<=n;i++){
            xor^=i;
        }
        return xor^xorArr;
    }
}

File: leetcode-solutions/0268-missing-number/solution.cpp
class Solution {
public:
    int missingNumber(vector<int>& arr) {
        int i=0;
        int n=arr.size();
        while(i<n){
            int correct=arr[i];
            if(correct < n && arr[i]!=arr[correct]){
                swap(arr[i],arr[correct]);
            }
            else i++;
        }
        for(int i=0;i<n;i++){
            if(arr[i]!=i) return i;
        }
        return n;
    }
};
